src/lib/widget/_utilities/component.utilities.ts

```diff
@@ -15,52 +16,67 @@ export interface ComponentStoreValue {
 }
 
 export const componentStore: Writable<ComponentStoreValue> = writable({
-  modal: null,
-  form: null,
+  lastAction: null,
   error: null,
+  form: null,
+  modal: null,
   mounted: false,
   open: null,
   reason: null,
   type: null,
 });
 
-export const componentApi = () => {
-  const { update, subscribe } = componentStore;
+/**
+ * @function closeComponent - this is a widget internal function not to be exposed to user
+ * @param {object} args - object containing  the reason for closing component
+ * @param {boolean} shouldCloseDialog - if true, the close command comes from outside of dialog component
+ */
+export function closeComponent(
+  args?: { reason: ComponentStoreValue['reason'] },
+  shouldCloseDialog?: boolean,
+) {
+  componentStore.update((state) => {
+    if (state.type === 'inline') {
+      console.warn('Component type of "inline" has no `close` method');
+      // There's nothing to do, so just return existing state
+      return state;
+    }
+    if (!state.modal?.component) {
+      console.warn('Modal component is not mounted. Please instantiate the Widget before use.');
+      // There's nothing to do, so just return existing state
+      return state;
+    }
 
-  return {
-    close: (args?: { reason: 'auto' | 'external' | 'user' }) => {
-      update((state) => {
-        if (state.type === 'inline') {
-          console.warn('Component type of "inline" has no `close` method');
-          // There's nothing to do, so just return existing state
-          return state;
-        }
-        if (!state.modal) {
-          console.warn('Modal component is not mounted. Please instantiate the Widget before use.');
-          // There's nothing to do, so just return existing state
-          return state;
-        }
+    shouldCloseDialog && state.modal.component.closeDialog();
 
-        state.modal?.component.closeDialog();
+    return {
+      ...state,
+      lastAction: 'close',
+      open: false,
+      reason: args?.reason || null,
+    };
+  });
+}
 
-        return {
-          ...state,
-          open: false,
-          reason: args?.reason || null,
-        };
-      });
-    },
-    mount: (component: SvelteComponent, element: HTMLDialogElement) => {
-      update((state) => {
-        return {
-          ...state,
-          modal: {
-            ...(component && { component, element }),
-          },
-          mounted: true,
-          type: component ? 'modal' : 'inline',
-        };
+export const componentApi = () => {
+  const { update } = componentStore;
+  // Create derived store to minimize what's exposed to the dev
+  const {
+    subscribe,
+  }: Readable<Pick<ComponentStoreValue, 'error' | 'lastAction' | 'mounted' | 'open' | 'reason'>> =
+    derived([componentStore], ([$componentStore], set) => {
```

Comment:
Moving to a derived store to only expose certain aspects to the component store.